main.go

```diff
@@ -73,30 +82,38 @@ func main() {
 	r.Handler("GET", *metricsPath, prometheus.Handler())
 
 	// Handlers for pushing and deleting metrics.
-	r.PUT("/metrics/job/:job/*labels", handler.Push(ms, true))
-	r.POST("/metrics/job/:job/*labels", handler.Push(ms, false))
-	r.DELETE("/metrics/job/:job/*labels", handler.Delete(ms))
-	r.PUT("/metrics/job/:job", handler.Push(ms, true))
-	r.POST("/metrics/job/:job", handler.Push(ms, false))
-	r.DELETE("/metrics/job/:job", handler.Delete(ms))
+	pushAPIPath := *routePrefix + "/metrics"
```

Comment:
At this point, we have already registered "/-/healthy", "/-/ready", and "/metrics". I think we want them prefixed as well. (At least that's what the Prometheus server is doing.) WDYT?

Reply:
> At this point, we have already registered "/-/healthy", "/-/ready", and "/metrics". I think we want them prefixed as well. (At least that's what the Prometheus server is doing.) WDYT?

I've prefixed `/-/*` with `*routePrefix`, but it seems like `*metricsPath` should be unaffected, since it's configured separately.

Reply:
Or we could prefix the configured metricsPath, or we can prefix it only if metricsPath is not configured explicitly. But in practice, we should just do what the Prometheus server is doing. Interestingly, at some point it was decided that we don't even need a `--web.telemetry-path` flag anymore for the Prometheus server. But before that flag was removed, the `--web.route-prefix` was used to prefix also the `--web.telemetry-path`. That's one point for doing it in the PGW, too. The other point is that we most likely will eventually remove `--web.telemetry-path` from the PGW, too, In that case, we would definitely need to prefix the (hardcoded) `/metrics` path. Which means we should do it now, too.

Long story short: Please prefix the metrics exposition path, too. Configured or not.